### PR TITLE
Karmeshyadav/robodev changes

### PR DESCRIPTION
**Pull request by ykarmesh**
## Motivation and Context

Changes from the time of the real world experiments for OVMM.

## How Has This Been Tested

Tested in the real world

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml [hunk 1]
```diff
@@ -13,13 +13,13 @@ habitat_baselines:
   rl:
     policy:
         name: "PointNavResNetPolicy"
-        action_distribution_type: "gaussian"
+        action_distribution_type: "categorical"
```
Comment (cpaxton) [inline projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml:16]: no continuous version still?

File: projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml [hunk 2]
```diff
@@ -13,13 +13,13 @@ habitat_baselines:
...
         no_downscaling: True
-        action_dist:
-            use_log_std: True
-            clamp_std: True
-            std_init: -1.0
-            use_std_param: True
+        #action_dist:
```
Comment (cpaxton) [inline projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml:18]: Why is this commented out? should we leave an explanation for these fields?

Comment (cpaxton) [inline projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml:18]: Why is this commented out? Should we add a description for these fields?

File: projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml (hunk 1 above)
Comment (cpaxton) [inline projects/habitat_ovmm/configs/agent/nav_to_obj_rl.yaml:16]: comment "gaussian or categorical" maybe?

File: projects/habitat_ovmm/configs/agent/floorplanner_eval.yaml [hunk 3]
```diff
@@ -120,20 +120,20 @@ AGENT:
...
+        - turn_right
       arm_joint_mask: [0, 0, 0, 0, 0, 0, 0] # the arm joints that the policy can control
       max_displacement: 0.25                # used when training the policy; could be different from the eval values
       max_turn_degrees: 30.0
       min_turn_degrees: 5.0
       min_displacement: 0.1
       sensor_height: 160
       sensor_width: 120
-      terminate_condition: continuous_stop
+      terminate_condition: discrete_stop
```
Comment (cpaxton) [inline projects/habitat_ovmm/configs/agent/floorplanner_eval.yaml:136]: we need to document what these options do

File: projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml [hunk 4]
```diff
@@ -14,8 +14,8 @@ ENVIRONMENT:
   frame_width: 480        # first-person frame width (in pixels)
   camera_height: 1.31     # camera sensor height (in metres)
   hfov: 42.0              # horizontal field of view (in degrees)
-  min_depth: 0.5          # minimum depth for depth sensor (in metres)
-  max_depth: 5.0         # maximum depth for depth sensor (in metres)
+  min_depth: 0.0          # minimum depth for depth sensor (in metres)
+  max_depth: 10.0         # maximum depth for depth sensor (in metres)
```
Comment (cpaxton) [inline projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml:18]: is this aligned with the robot?

File: projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml [hunk 5]
```diff
@@ -51,7 +51,7 @@ AGENT:
   SKILLS:
     GAZE_OBJ:
       type: rl #end_to_end #heuristic #hardcoded
-      checkpoint_path: data/checkpoints/ovmm/gaze_may18_small_img_80%.pth
+      checkpoint_path: data/checkpoints/ovmm/May31/cat_gaze_recep_sensor_200m_may31.pth
```
Comment (cpaxton) [inline projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml:54]: are these checkpoints uploaded?

File: projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml [hunk 6]
```diff
@@ -95,25 +95,25 @@ AGENT:
...
       sensor_height: 160
       sensor_width: 120
-      terminate_condition: continuous_stop
+      terminate_condition: discrete_stop #continuous_stop
       nav_goal_seg_channels: 2
 
     NAV_TO_REC:
-      type: heuristic # heuristic (default) or rl
-      checkpoint_path: data/checkpoints/ovmm/nav_to_rec_may26_teleport_dist_0_1_77%.pth
+      type: rl # heuristic (default) or rl
```
Comment (cpaxton) [inline projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml:115]: Should we just create separate config files for RL vs Heuristic eval and pass them in on the command line?

File: projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml [hunk 7]
```diff
@@ -124,32 +124,32 @@ AGENT:
...
-      type: heuristic # "rl" or "heuristic" or "hardcoded"
-      checkpoint_path: data/checkpoints/ovmm/place_with_manip_action_may25.pth
+      type: rl # "rl" or "heuristic" or "hardcoded"
+      checkpoint_path: data/checkpoints/ovmm/May31/cat_place_kill__before_20_785_tilt_200m_may31.pth
       rl_config: projects/habitat_ovmm/configs/agent/place_rl.yaml # with continuous actions
       gym_obs_keys:
         - robot_head_depth
         - goal_receptacle
         - joint
-        - ovmm_nav_goal_segmentation
```
Comment (cpaxton) [inline projects/stretch_ovmm/configs/agent/floorplanner_eval.yaml:152]: not sure I understand this change

File: src/home_robot/home_robot/motion/stretch.py [hunk 8]
```diff
@@ -39,6 +39,23 @@
     ]
 )
 
+# look down in navigation mode for doing manipulation post-navigation
+STRETCH_POSTNAV_Q = np.array(
```
Comment (cpaxton) [inline src/home_robot/home_robot/motion/stretch.py:43]: wasn't this already added?

File: src/home_robot_hw/home_robot_hw/env/visualizer.py [hunk 9]
```diff
@@ -251,7 +268,7 @@ def visualize(
 
     def _init_vis_image(self, goal_name: str):
         # vis_image = np.ones((655, 1165, 3)).astype(np.uint8) * 255
-        vis_image = np.ones((655, 885, 3)).astype(np.uint8) * 255
+        vis_image = np.ones((655, 1600, 3)).astype(np.uint8) * 255
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/visualizer.py:271]: can you verify visualizations still look right?

File: src/home_robot_hw/home_robot_hw/env/visualizer.py [hunk 10]
```diff
@@ -460,7 +477,7 @@ def visualize(
             )
 
     def _init_vis_image(self, goal_name: str):
-        vis_image = np.ones((655, 1165, 3)).astype(np.uint8) * 255
+        vis_image = np.ones((655, 2665, 3)).astype(np.uint8) * 255
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/visualizer.py:480]: same - maybe add screenshots?

File: src/home_robot_hw/home_robot_hw/remote/modules/head.py [hunk 11]
```diff
@@ -112,8 +112,8 @@ def get_images(self, compute_xyz=False):
     # Helper methods
 
     def _process_depth(self, depth):
-        depth[depth < self.min_depth_val] = MIN_DEPTH_REPLACEMENT_VALUE
-        depth[depth > self.max_depth_val] = MAX_DEPTH_REPLACEMENT_VALUE
+        # depth[depth < self.min_depth_val] = MIN_DEPTH_REPLACEMENT_VALUE
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/remote/modules/head.py:115]: why is this function even called?

File: src/home_robot_sim/home_robot_sim/env/habitat_objectnav_env/visualizer.py [hunk 12]
```diff
@@ -356,6 +357,24 @@ def visualize(
             color = self.semantic_category_mapping.map_color_palette[9:12][::-1]
             cv2.drawContours(image_vis, [agent_arrow], 0, color, -1)
 
+        if rl_obs_frame is not None:
```
Comment (cpaxton) [inline src/home_robot_sim/home_robot_sim/env/habitat_objectnav_env/visualizer.py:360]: check visualizations?

**Review by cpaxton — CHANGES_REQUESTED** [review]
Some questions + want to make sure we commit working config files

**Comment by ykarmesh** [comment]
Looks like all the changes from `robodev` were separately done on `main` by Sriram. Closing this PR

**Pull request closed without merging**